src/main.cpp

```diff
@@ -4253,6 +4258,18 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     return true;
 }
 
+static bool CheckMessageHeader(CNode* pnode, const CMessageHeader* msg)
+{
+    if (msg->nMessageSize > MAX_PROTOCOL_MESSAGE_LENGTH) {
+        LOCK(cs_main);
+        Misbehaving(pnode->GetId(), 100);
+        LogPrintf("net", "Oversized message from peer=%d (size=%d)\n", pnode->GetId(), msg->nMessageSize);
```

Comment:
I lobbed a 4MB message at my node and it crashed here:
```
EXCEPTION: St13runtime_error       
tinyformat: Not enough conversion specifiers in format string       
bitcoin in net       
```
As a category is passed, this should be `LogPrint`, not `LogPrintf`.